app/assets/javascripts/documents.js

```diff
@@ -0,0 +1,63 @@
+$(".document-input").change(documentsChanged);
+$("#post_title_en").change(updateFileUrls);
+$("#page_title").change(updateFileUrls);
+function convertToSlug(Text)
+{
+    return Text
+        .toLowerCase()
+        .replace(/[^\w ]+/g,'')
+        .replace(/ +/g,'-')
+        ;
+}
+function documentsChanged(){
+  updateFileUrls();
+}
+function updateFileUrls(){
+  var div = document.getElementById("documents_container");
+  var elements = div.getElementsByTagName("input");
+  var fileList = []
+  for (var i = 0; i < elements.length; i += 1) {
+    for (var j = 0; j < elements[i].files.length; j += 1) {
+      fileList.push(elements[i].files[j]);
+    }
+  }
+
+  var ul = document.getElementById("list-document-urls");
+  try {
+    var trans = I18n.t("activerecord.attributes.post.title");
+  } catch (e){
+    var trans = "Title";
+  }
+  $('.document-url').remove()
+  if(document.getElementById("post_title_en") != null){
```

Comment:
Do this calculation (with try-catch) once, not every time the user changes the title?